src/native_main.nim

```diff
@@ -137,6 +137,16 @@ proc handleMessage(msg: MessageRecv): MessageResp =
             result.code = some waitForExit(process)
             close(process)
 
+        of "run_async":
+            when defined(windows):
+                let command = "cmd /c " & msg.command.get()
+                createOrphanProcess(command)
+            else:
+                let command = msg.command.get()
+                discard startProcess(command, options = {poEvalCommand})
```

Comment:
is this definitely `async` enough that it can be lives after native exits?

Reply:
I'm not completely sure. [The doc](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_messaging#closing_the_native_app) says you have to put the processes you want preserved into their own process group, but maybe the program I tested it with (`gnome_terminal`) does that anyway. I'll look into it more.

Reply:
OK, I'm confused now. I took another look at `poDaemon` and it's *supposed to* make the process run in its own process group – which it does, but only when I use `gen_native_message.py` and not when I try it through Firefox. But either way, the process keeps running???

Reply:
> and not when I try it through Firefox

I'd check that you're _definitely_ calling the version you think you're calling through Firefox before you spend too long debugging it.

Reply:
No, I'm absolutely sure. I've now tried it on both WSL2 and in a Lubuntu VM, and it looks like `poDaemon` has no effect at all. When the native messenger is called manually, a new process group is created, and when it's called by Firefox, it isn't. Either way, the process doesn't get killed by Firefox.

Reply:
In retrospect I think we already knew the process wouldn't be killed since `:restart` works easily on Linux.